src/rdb.c

```diff
@@ -2541,7 +2570,18 @@ int rdbLoadRio(rio *rdb, int rdbflags, rdbSaveInfo *rsi) {
          * Similarly if the RDB is the preamble of an AOF file, we want to
          * load all the keys as they are, since the log of operations later
          * assume to work in an exact keyspace state. */
-        if (iAmMaster() &&
+        if (val == NULL) {
+            /* Note: Since we used to have bug that could lead to create empty
+             * keys, so don't fail when empty key is encountered, we will
+             * silently discard it and continue loading. See #8453. */
```

Comment:
```suggestion
            /* Since we used to have bug that could lead to empty keys
             * (See #8453), we rather not fail when empty key is encountered
             * in an RDB file, instead we will silently discard it and
             * continue loading. */
```